### PR TITLE
Corrected a namespace error in argparse call throwing namespace error

### PR DESCRIPTION
**Pull request by puter**

### CODE REVIEW

**Comment by tupton** [comment]
What's the error that you're seeing?

File: weather.py [hunk 1]
```diff
@@ -206,39 +206,40 @@ def create_cli_parser():
...
 def main(argv):
-    """
-    Main entry point of this file. Parses argv, gets weather, then emits output
-    """
-
-    # Create the command line parser.
-    cli_parser = create_cli_parser()
-
-    # Get the options and arguments.
-    opts, args = cli_parser.parse_args(argv)
```
Comment (tupton) [inline weather.py:217]: The [docs for optparse](https://docs.python.org/2/library/optparse.html) indicate that `parse_args` returns a 2-tuple of options and arguments.

``` python
from optparse import OptionParser
[...]
parser = OptionParser()
parser.add_option("-f", "--file", dest="filename",
                  help="write report to FILE", metavar="FILE")
parser.add_option("-q", "--quiet",
                  action="store_false", dest="verbose", default=True,
                  help="don't print status messages to stdout")

(options, args) = parser.parse_args()
```

File: weather.py [hunk 2]
```diff
@@ -206,39 +206,40 @@ def create_cli_parser():
...
+	# Get the options and arguments.
+	opts = cli_parser.parse_args(argv)
+
+	# Limit the requested forecast days.
+	if opts.forecast > DAYS_LIMIT or opts.forecast < 0:
+		cli_parser.error("Days to forecast must be between 0 and %d"
+						 % DAYS_LIMIT)
+
+	# Get the weather.
+	# weather = get_weather(args.location_code, opts)
```
Comment (tupton) [inline weather.py]: No need to leave a commented-out line in the code.

**Comment by puter** [comment]
after a fresh clone, the error I'm seeing is: 
~/python-yahoo-weather$ ./weather.py 10101
Traceback (most recent call last):
  File "./weather.py", line 244, in <module>
    main(sys.argv[1:])
  File "./weather.py", line 217, in main
    opts, args = cli_parser.parse_args(argv)
TypeError: 'Namespace' object is not iterable

Simply pulling everything into one var seemed to fix the issue. I check the rest of the code for any potential conflicts and saw none. Each separate key in the namespace is called uniquely, so keeping your arg with the rest of the opts seems not to negatively affect the rest of the project.

**Comment by puter** [comment]
Also, I believe you're using the argparse module here: https://docs.python.org/2/library/argparse.html#module-argparse

And not the optparse module whose doc you linked me to.

**Comment by tupton** [comment]
Oof, right you are. I switched to `argparse` a few months ago but I must not have tested properly. For some reason I wasn't up to date with my local copy and didn't see this error. Thanks for catching.

**Comment by puter** [comment]
NP. Thanks for the project.

**Pull request closed without merging**